### PR TITLE
moving concepts to concepts:: namespace

### PR DESCRIPTION
**Pull request by dcoeurjo**
# PR Description

Fixing issue #1189 

# Checklist

- [x] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [x] Doxygen documentation of the code completed (classes, methods, types, members...)
- [x] Documentation module page added or updated.
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).
- [x] All continuous integration tests pass (Travis & appveyor)

### CODE REVIEW

File: src/DGtal/geometry/curves/CBidirectionalSegmentComputer.h [hunk 1]
```diff
@@ -47,55 +47,56 @@
...
   public:
     // Methods
     BOOST_CONCEPT_USAGE( CBidirectionalSegmentComputer )
     {
-      concepts::ConceptUtils::sameType( myB, myX.isExtendableBack() );
-      concepts::ConceptUtils::sameType( myB, myX.extendBack() );
-    }
+    concepts::ConceptUtils::sameType( myB, myX.isExtendableBack() );
+    concepts::ConceptUtils::sameType( myB, myX.extendBack() );
+  }
```
Comment (rolanddenis) [inline src/DGtal/geometry/curves/CBidirectionalSegmentComputer.h]: There is an indentation problem here (the {}).

File: src/DGtal/geometry/curves/CBidirectionalSegmentComputer.h [hunk 2]
```diff
@@ -105,11 +106,11 @@ Description of \b concept '\b CBidirectionalSegmentComputer' <p>
   private:
     
   }; // end of concept CBidirectionalSegmentComputer
-  
-} // namespace DGtal
+  }
```
Comment (rolanddenis) [inline src/DGtal/geometry/curves/CBidirectionalSegmentComputer.h]: missing a `// namespace concepts`

File: src/DGtal/geometry/curves/CBidirectionalSegmentComputer.h [hunk 3]
```diff
@@ -105,11 +106,11 @@ Description of \b concept '\b CBidirectionalSegmentComputer' <p>
...
+  } // namespace DGtal
 
-//                                                                           //
-///////////////////////////////////////////////////////////////////////////////
+    //                                                                           //
+    ///////////////////////////////////////////////////////////////////////////////
```
Comment (rolanddenis) [inline src/DGtal/geometry/curves/CBidirectionalSegmentComputer.h]: un-indent ?

File: src/DGtal/base/CStack.h [hunk 4]
```diff
@@ -47,77 +47,78 @@
...
+      }
+      // ------------------------- Private Datas --------------------------------
+    private:
+      T myX; // do not require T to be default constructible.
+      Value myV; 
+      Size myS; 
+      bool myB; 
+
+    }; // end of concept CStack
+  }
```
Comment (rolanddenis) [inline src/DGtal/base/CStack.h]: `// namespace concepts`

File: src/DGtal/geometry/curves/CDynamicBidirectionalSegmentComputer.h [hunk 5]
```diff
@@ -79,8 +80,8 @@ Description of \b concept '\b CDynamicBidirectionalSegmentComputer' <p>
    */
   template <typename T>
   struct CDynamicBidirectionalSegmentComputer :
-    CBidirectionalSegmentComputer<T>,
-    CDynamicSegmentComputer<T>
+    concepts::CBidirectionalSegmentComputer<T>,
+    concepts::CDynamicSegmentComputer<T>
```
Comment (rolanddenis) [inline src/DGtal/geometry/curves/CDynamicBidirectionalSegmentComputer.h]: Was not necessary but OK.

File: src/DGtal/geometry/curves/CDynamicBidirectionalSegmentComputer.h [hunk 6]
```diff
@@ -97,7 +98,7 @@ Description of \b concept '\b CDynamicBidirectionalSegmentComputer' <p>
     // ------------------------- Internals ------------------------------------
   private:
   }; // end of concept CDynamicBidirectionalSegmentComputer
-
+ }
```
Comment (rolanddenis) [inline src/DGtal/geometry/curves/CDynamicBidirectionalSegmentComputer.h]: `// namespace concepts` (but OK, you will maybe not update all the 50 files again)

File: src/DGtal/geometry/curves/CForwardSegmentComputer.h [hunk 7]
```diff
@@ -47,69 +47,70 @@
...
+       @code
+       for ( ConstIterator it = s.begin(),
+       ConstIterator itEnd = s.end();
+       it != itEnd; ++it)
+       { // [it, itEnd) is a segment:
+       bool flag = true;
+       while ( (s.end() != itEnd)&&(flag) ) { flag = s.extend(); }
+       ASSERT( flag );
+       }
+       @endcode
```
Comment (rolanddenis) [inline src/DGtal/geometry/curves/CForwardSegmentComputer.h]: Could you re-indent the `@code` part as before ?

File: src/DGtal/geometry/curves/CIncrementalSegmentComputer.h [hunk 8]
```diff
@@ -47,90 +47,92 @@
...
+       for ( ConstIterator it = s.begin(), 
+       ConstIterator itEnd = s.end();
+       it != itEnd; ++it)
+       { // [s.begin(), it) is a segment:
+       s.init( s.begin() ); 
+       bool flag = true; 
+       while ( (s.end() != it)&&(flag) ) { flag = s.extend(); }
+       ASSERT( flag ); 
+       }
+       @endcode
```
Comment (rolanddenis) [inline src/DGtal/geometry/curves/CIncrementalSegmentComputer.h]: Same here (code indentation).

File: src/DGtal/geometry/curves/CDynamicBidirectionalSegmentComputer.h (hunk 6 above)
Comment (dcoeurjo) [inline src/DGtal/geometry/curves/CDynamicBidirectionalSegmentComputer.h]: > Le 24 juin 2016 à 19:36, rolanddenis <notifications@github.com> a écrit :
> 
> In src/DGtal/geometry/curves/CDynamicBidirectionalSegmentComputer.h:
> 
> > @@ -97,7 +98,7 @@ Description of \b concept '\b CDynamicBidirectionalSegmentComputer' <p>
> >      // ------------------------- Internals ------------------------------------
> >    private:
> >    }; // end of concept CDynamicBidirectionalSegmentComputer
> > -
> > + }
> // namespace concepts (but OK, you will maybe not update all the 50 files again)
> 
Yes, please 😉

File: src/DGtal/geometry/curves/estimation/CCurveLocalGeometricEstimator.h [hunk 9]
```diff
@@ -111,12 +112,12 @@ namespace DGtal
...
-      myX.init( myH, myItb, myIte ); 
+    //init method
+    myX.init( myH, myItb, myIte ); 
 
-      concepts::ConceptUtils::sameType( myQ, myX.eval( myItb ) );
-      concepts::ConceptUtils::sameType( myIto, myX.eval( myItb, myIte, myIto ) );
-    }
+    concepts::ConceptUtils::sameType( myQ, myX.eval( myItb ) );
+    concepts::ConceptUtils::sameType( myIto, myX.eval( myItb, myIte, myIto ) );
+  }
```
Comment (rolanddenis) [inline src/DGtal/geometry/curves/estimation/CCurveLocalGeometricEstimator.h]: There is an indentation problem inside the `BOOST_CONCEPT_USAGE` block.

File: src/DGtal/geometry/curves/estimation/CCurveLocalGeometricEstimator.h [hunk 10]
```diff
@@ -133,11 +134,11 @@ namespace DGtal
...
   }; // end of concept CCurveLocalGeometricEstimator
+  }
+  } // namespace DGtal
 
-} // namespace DGtal
-
-//                                                                           //
-///////////////////////////////////////////////////////////////////////////////
+    //                                                                           //
+    ///////////////////////////////////////////////////////////////////////////////
```
Comment (rolanddenis) [inline src/DGtal/geometry/curves/estimation/CCurveLocalGeometricEstimator.h]: Indentation.

File: src/DGtal/io/CDrawableWithDisplay3D.h [hunk 11]
```diff
@@ -48,83 +48,84 @@
...
-   \code
-   Display3D display;
-   display << CustomStyle( x.className(), x.defaultStyle() )
-         << x;
-   \endcode 
+       \code
+       Display3D display;
+       display << CustomStyle( x.className(), x.defaultStyle() )
+       << x;
+       \endcode
```
Comment (rolanddenis) [inline src/DGtal/io/CDrawableWithDisplay3D.h]: `@code` indentation.

File: src/DGtal/io/boards/CDrawableWithBoard2D.h [hunk 12]
```diff
@@ -52,79 +52,81 @@ namespace DGtal
...
-   \code
-   Board2D board;
-   board << CustomStyle( x.className(), x.defaultStyle() )
-         << x;
-   \endcode 
+       \code
+       Board2D board;
+       board << CustomStyle( x.className(), x.defaultStyle() )
+       << x;
+       \endcode
```
Comment (rolanddenis) [inline src/DGtal/io/boards/CDrawableWithBoard2D.h]: `@code` indentation.

File: src/DGtal/io/boards/CDrawableWithBoard3DTo2D.h [hunk 13]
```diff
@@ -47,89 +47,90 @@
...
-   \code
-   Board3DTo2D display;
-   display << CustomStyle( x.className(), x.defaultStyle() )
-         << x;
-   \endcode 
+       \code
+       Board3DTo2D display;
+       display << CustomStyle( x.className(), x.defaultStyle() )
+       << x;
+       \endcode
```
Comment (rolanddenis) [inline src/DGtal/io/boards/CDrawableWithBoard3DTo2D.h]: `@code` indentation

File: src/DGtal/io/boards/CDrawableWithBoard3DTo2D.h [hunk 14]
```diff
@@ -47,89 +47,90 @@
...
-} // namespace DGtal
+  }
+  } // namespace DGtal
 
 
 
-//                                                                           //
-///////////////////////////////////////////////////////////////////////////////
+    //                                                                           //
+    ///////////////////////////////////////////////////////////////////////////////
```
Comment (rolanddenis) [inline src/DGtal/io/boards/CDrawableWithBoard3DTo2D.h]: Unnecessary indent.

File: src/DGtal/io/viewers/CDrawableWithViewer3D.h [hunk 15]
```diff
@@ -47,77 +47,78 @@
...
-   \code
-   Viewer3D display;
-   display << CustomStyle( x.className(), x.defaultStyle() )
-         << x;
-   \endcode 
+       \code
+       Viewer3D display;
+       display << CustomStyle( x.className(), x.defaultStyle() )
+       << x;
+       \endcode
```
Comment (rolanddenis) [inline src/DGtal/io/viewers/CDrawableWithViewer3D.h]: `@code` indent.

**Comment by rolanddenis** [comment]
Minor remarks about indentation but great job !

**Comment by dcoeurjo** [comment]
all done

**Comment by dcoeurjo** [comment]
waiting for the buildbots

**Comment by rolanddenis** [comment]
Fine, merging !

**Pull request merged**